### DISCUSSION TITLE
How would I add a scale bar to images?

### DISCUSSION THREAD

**Discussion by afvrbanac** [histolab/histolab#293] [Q&A] [answered]
Hello, novice user here with some basic questions. I'm using histolab to extract random tiles at different levels. Is it possible to automatically add a scale bar to these images? 

Also, is the magnification of each level the original magnification divided by the downsample for the level? So for example, for a 40X image at level 0, if the downsample for level 2 was 16, the magnification would be 2.5X?

**Comment by ernestoarbitrio** [comment]
Hi @afvrbanac I will transfer this issue in a github discussion!!! 😄

**Comment by nicolebussola** [comment, marked as answer]
Hi Alison, thank you for your questions, absolutely not trivial. Regarding the possibility to add a scale bar, we haven't thought of that as   it is not optimal to have that if you want to use the tiles in a machine learning/deep learning pipeline. We can try and add the scalar bar as an optional parameter when saving the tiles (do you have specific needs?). 

Regarding the magnification level, your guess is correct; I have checked with an expert pathologist on a few samples and the conversion holds, so you can do the math by reading the downsample factor for each level by using the ``.properties`` method of the `slide`. I think that this conversion might be a nice feature to have so we will definitely add this on our next release, so thank you 😄 
Best
Nicole

**Reply by afvrbanac** [reply]
Thanks Nicole, this is very helpful! Yeah I'll be using the tiles both for machine learning purposes and they will also be manually annotated in ImageJ for comparison (the annotator had requested a scale bar which is why I was asking).
